src/components/plots/rectanglePlot.ts

```diff
@@ -63,10 +63,10 @@ export module Plots {
       return [{attrToProjector: this._generateAttrToProjector(), animator: this._getAnimator("rectangles")}];
     }
 
-    public x1(): AccessorScaleBinding<X, number>;
-    public x1(x1: number | _Accessor): Plots.Rectangle<X, Y>;
-    public x1(x1: X | _Accessor, scale: Scale<X, number>): Plots.Rectangle<X, Y>;
-    public x1(x1?: number | _Accessor | X, scale?: Scale<X, number>): any {
+    public x1<X1>(): AccessorScaleBinding<X1, number>;
+    public x1(x1: number | Accessor<number>): Plots.Rectangle<X, Y>;
+    public x1<X1>(x1: X1 | Accessor<X1>, scale: Scale<X1, number>): Plots.Rectangle<X, Y>;
+    public x1<X1>(x1?: number | Accessor<number> | X1 | Accessor<X1>, scale?: Scale<X1, number>): any {
```

Comment:
Why `<X1>` and `<X2>`? Don't they have to be the same type as the generic `<X>` on the `Plots.Rectangle`?